cpp/tests/test_abm.cpp

```diff
@@ -1481,3 +1420,148 @@ TEST(TestDiscreteDistribution, generate)
         ASSERT_LE(d, 4);
     }
 }
+
+TEST(TestTestingRule, addremoveandevaluateTestRule)
+{
+    auto world   = mio::abm::World();
+    auto home_id = world.add_location(mio::abm::LocationType::Home);
+    auto work_id = world.add_location(mio::abm::LocationType::Work);
+    auto person  = mio::abm::Person(home_id, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age15to34,
+                                    world.get_global_infection_parameters());
+    auto& home   = world.get_individualized_location(home_id);
+    auto& work   = world.get_individualized_location(work_id);
+    person.set_assigned_location(home);
+    person.set_assigned_location(work);
```

Comment:
we should avoid using unnecessary objects in tests. world is not needed here. directly create the location.